skbeam/core/accumulators/binned_statistic.py

```diff
@@ -316,9 +318,12 @@ def __call__(self, values, statistic=None):
                     null = np.nan
                 np.seterr(**old)
             self.result.fill(null)
-            for i in np.unique(self.xy):
+            p = Pool()
+
+            def op(i):
                 self.result[i] = statistic(values[self.xy == i])
 
+            [_ for _ in p.imap_unordered(op, np.unique(self.xy), 10)]
```

Comment:
We get away with this because: 1) numpy releases the GIL (most of the time), 2) we know that each process will be touching a separate element in `self.result` so it is implicitly thread safe 3) all the important players are in global memory. We could move this to multiprocessing to avoid the GIL entirely but 1) processes take more time to spin up/manage 2) we would need to declare the shared nature of the memory. If someone has strong opinions/experience with this please let me know if one way is better than the other.

Reply:
I do not understand the second (2) about 'declaring the shared nature of the memory'

Reply:
We would need to have some sort of shared array object or give each process a copy of the array, which is not a problem in threading. https://docs.python.org/2/library/multiprocessing.html#sharing-state-between-processes